spec.md

```diff
@@ -2134,6 +2134,19 @@ message NodeStageVolumeRequest {
   // This field is OPTIONAL and MUST match the volume_context of the
   // volume identified by `volume_id`.
   map<string, string> volume_context = 6;
+
```

Comment:
this field is declared both for node-stage, and for node-publish. why not declare it ONCE as part of `VolumeCapability` ?

Reply:
or, if this is only for mount .. then perhaps as part of `MountVolume`?

Reply:
and what does this field mean in the context of block vols?

Reply:
It is not possible to use mount flags defined in volume capability (inside `MountVolume`) because:

1. The flags defined inside volume capability is static and part of `CreateVolume` response - whereas this mount option depends on workload that is trying to use the volume and could change.
2. Say even if we pretend that volume capability can be modified at any point in time and is not tied to `Volume` object. How does CO know it should supply `-o gid=<>` mount flag during NodeStage or NodePublish?

Reply:
hmm - I think your response meant the flag not be included as `mount_flags` inside `MountVolume` type - but rather a separate field in `MountVolume` type? I misunderstood it to be the former. 

Yes having a separate `volume_mount_group` field inside `MountVolume` could work. It would have same semantic meaning as when it is defined outside (as current PR). The main concern though is - it would perhaps alter meaning of volume capability. But if that is acceptable trade-off I can make that change.

Reply:
I have moved this field to `VolumeCapability` with everything else being the same. I think this also allows us to cleanup the error handling and now error handling simply is determined by based on the criteria if SP supports publishing a volume with different capabilities on same node.  It also addresses interaction of this PR with https://github.com/container-storage-interface/spec/pull/476